WordPress/Classes/Services/LikeUserHelpers.swift

```diff
@@ -59,11 +59,8 @@ import CoreData
     }
 
     class func purgeStaleLikes() {
-        let derivedContext = ContextManager.shared.newDerivedContext()
-
-        derivedContext.perform {
-            purgeStaleLikes(fromContext: derivedContext)
-            ContextManager.shared.save(derivedContext)
+        ContextManager.shared.save {
```

Comment:
This is IMHO the ideal replacement strategy – tiny changes that don't adjust the overall approach or semantics of the operation, just consolidate the complexity into the `ContextManager`. It's definitely not as forward-thinking, but it's possible to know just by looking at it that it's safe.